### DISCUSSION TITLE
How are images resized

### DISCUSSION THREAD

**Discussion by an3252sv** [ultralytics/yolov5#7126] [Q&A] [answered]
I have a data set with 1280x720 sized images. If I set the network parameter --img-size=640, how will my images be resized? I have read that the images aspect ratio will be preserved i.e, they get downscaled to 640x360 (same aspect ratio as 1280x720), then padding is added to make the image 640x640. Source: https://github.com/pjreddie/darknet/issues/800, written by 'pullmyleg' on Apr 6 2020. Their answer refers to yolov3, but is the same method used for yolov5? I would really like to be pointed to the resource where this is stated.

**Comment by glenn-jocher** [comment, marked as answer]
@an3252sv images will be resized to --img-size on the long side, short side handled automatically preserving aspect ratio (circles stay circles). Letterboxing (grey padding) is used to fullfill stride constraints (i.e. 32 or 64 pixel multiples).

**Reply by naik-amey** [reply]
Does resizing help make better detections? Why not keep the image as it is and pad it to input size. 
For example: let's say the image size is 224x224 and yolov5 input size is 640x640, can we skip resizing and pad it to form 640x640?
